parser/parser/cc/grammars/typedruby.ypp

```diff
@@ -1296,100 +1304,197 @@ lbrace_cmd_block_start:
+                | tUMINUS error
+                    {
+                      $$ = driver.build.unary_op(self, $1, driver.build.error_node(self, @1.end, @2.end));
+                    }
                 | arg tPIPE arg
                     {
                       $$ = driver.build.binaryOp(self, $1, $2, $3);
                     }
+                | arg tPIPE error
+                    {
+                      $$ = driver.build.binaryOp(self, $1, $2, driver.build.error_node(self, @2.end, @3.end));
+                    }
                 | arg tCARET arg
                     {
                       $$ = driver.build.binaryOp(self, $1, $2, $3);
                     }
+                | arg tCARET error
+                    {
+                      $$ = driver.build.binaryOp(self, $1, $2, driver.build.error_node(self, @2.end, @3.end));
+                    }
                 | arg tAMPER2 arg
                     {
                       $$ = driver.build.binaryOp(self, $1, $2, $3);
                     }
+                | arg tAMPER2 error
+                    {
+                      $$ = driver.build.binaryOp(self, $1, $2, driver.build.error_node(self, @2.end, @3.end));
+                    }
                 | arg tCMP arg
                     {
                       $$ = driver.build.binaryOp(self, $1, $2, $3);
                     }
+                | arg tCMP error
+                    {
+                      $$ = driver.build.binaryOp(self, $1, $2, driver.build.error_node(self, @2.end, @3.end));
+                    }
                 | rel_expr %prec tCMP
                 | arg tEQ arg
                     {
                       $$ = driver.build.binaryOp(self, $1, $2, $3);
                     }
+                | arg tEQ error
+                    {
+                      $$ = driver.build.binaryOp(self, $1, $2, driver.build.error_node(self, @2.end, @3.end));
+                    }
                 | arg tEQQ arg
                     {
                       $$ = driver.build.binaryOp(self, $1, $2, $3);
                     }
+                | arg tEQQ error
+                    {
+                      $$ = driver.build.binaryOp(self, $1, $2, driver.build.error_node(self, @2.end, @3.end));
+                    }
                 | arg tNEQ arg
                     {
                       $$ = driver.build.binaryOp(self, $1, $2, $3);
                     }
+                | arg tNEQ error
+                    {
+                      $$ = driver.build.binaryOp(self, $1, $2, driver.build.error_node(self, @2.end, @3.end));
+                    }
                 | arg tMATCH arg
                     {
                       $$ = driver.build.match_op(self, $1, $2, $3);
                       DIAGCHECK();
                     }
+                | arg tMATCH error
+                    {
+                      $$ = driver.build.match_op(self, $1, $2, driver.build.error_node(self, @2.end, @3.end));
+                      DIAGCHECK();
```

Comment:
This DIAGCHECK macro mentions YYERROR. I don't really know what the ramifications of this are. Do you know? Is this ok to put in a clause that mentions `error`?

Reply:
Nope, good question! Let me look into it.

Reply:
So I was looking into why the `DIAGCHECK` is here in the first place. I think it's because the original typedruby parser can produce a diagnostic when parsing the regexp: https://github.com/typedruby/typedruby/blob/0216fe987e1bfc55fce36f8c859d529aa9b8d870/parser/src/builder.rs#L286-L301 .

We don't actually implement this behavior, which I assume is why we wound up deleting the `InvalidRegexp` diagnostic in #5186. So presumably we don't technically need the original `DIAGCHECK` anyway. So I'm inclined to delete this one, even though I don't think mixing `error` with `YYERROR` is actually a problem.

Reply:
https://github.com/sorbet/sorbet/commit/792d36670e25289db5b705713796b19555b6a966 removes the `DIAGCHECK` and explains why.